### PR TITLE
Add classes for teachers and students

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
-In this PR I added classes to represent the teachers and students using OOP 

**Below are the highlights**

- Setup linters
- Use encapsulation and abstraction in the Person class
- Use Inheritance in creating the student and teacher class
- Use Polymorphism in creating some methods

**Cheers and Happy Reviewing!**

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age > 18
+  end
```
Comment (ShahierNashaat) [inline person.rb]: - [ ] Kindly notice that in the `of_age?` method you should check if the `age` is greater than or equal to 18 as required.

```ruby
def of_age?
    @age >= 18
end
```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom, _name = 'unknown')
+    super('sharon', 20)
+    @classroom = classroom
+  end
```
Comment (ShahierNashaat) [inline student.rb]: - [ ] Kindly notice that in the `Student` constructor you should send the data for the `Person` class constructor as it is its parent class.

```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
end
```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization, _name = 'unknown')
+    super('Mary', 17)
+    @specialization = specialization
+  end
```
Comment (ShahierNashaat) [inline teacher.rb]: - [ ] Also here in the `Teacher` constructor you should send the data for the `Person` class constructor as it is its parent class.

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
end
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @sharon-odhiambo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sharon-odhiambo** [comment]
Hi @ShahierNashaat , thank you for the review and suggestions. I've implemented them.

**Review by mikethreels — APPROVED** [review]
Hi Sharon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**